### PR TITLE
feat: Surface using logfmt for dev-time logs

### PR DESCRIPTION
**Pull request by plisy**
## What this PR does / why we need it

Adding both tee to the log file (so the logs show up in Datadog) and `logfmt` for easier to parse dev-time logs (there's a lot of details a dev doesn't need in the default log format).

For more info on the `logfmt` go here https://outreach-io.atlassian.net/wiki/spaces/EN/pages/706347415/Logging#Formatting-option

## Jira ID

[DTSS-1937]

## Notes for your reviewers

[DTSS-1937]: https://outreach-io.atlassian.net/browse/DTSS-1937?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: root/Makefile [hunk 1]
```diff
@@ -34,6 +34,7 @@ FLY                 ?= $(shell ./scripts/shell-wrapper.sh gobin.sh -p github.com
 BENCH_FLAGS         := "-bench=Bench $(BENCH_FLAGS)"
 TEST_TAGS           ?= or_test,or_int
 SKIP_VALIDATE       ?=
+LOGFMT			    ?=
```
Comment (jaredallard) [inline root/Makefile:37]: nit: mixing tabs and space

File: root/Makefile [hunk 2]
```diff
@@ -147,10 +149,16 @@ gobuild: check-deps
...
+	@DEVBOX_LOGFMT="$(LOGFMT)" ./scripts/shell-wrapper.sh air-runner.sh
+
 ## dev:       run the service
 .PHONY: dev
 dev:: pre-dev
-	@./scripts/shell-wrapper.sh gobin.sh github.com/cosmtrek/air@f7ec5b5ac29540748772dd4deb24a39f81ba9a47 -c $(CURDIR)/.air.toml
+	@DEVBOX_LOGFMT="$(LOGFMT)" ./scripts/shell-wrapper.sh gobin.sh github.com/cosmtrek/air@f7ec5b5ac29540748772dd4deb24a39f81ba9a47 -c $(CURDIR)/.air.toml
 
 ## devserver:       run the service
 .PHONY: devserver
```
Comment (jaredallard) [inline root/Makefile:164]: Should we just alias `devserver` to `run` right now?

File: shell/air-runner.sh [hunk 3]
```diff
@@ -11,5 +11,28 @@ APPNAME="$(get_app_name)"
...
+    if [[ -n $LOGFMT_FORMAT ]]; then
+      logfmt+=(--format "$LOGFMT_FORMAT")
+    fi
+
+    if [[ -n $LOGFMT_FILTER ]]; then
+      logfmt+=(--filter "$LOGFMT_FILTER")
+    fi
+
+    exec "$(get_repo_directory)/bin/$DEV_CONTAINER_EXECUTABLE" "$@" |
+      tee -ai "${DEV_CONTAINER_LOGFILE:-/tmp/app.log}" |
```
[the remark this answers is not in the document]
Comment (jaredallard) [reply shell/air-runner.sh:35]: Also, unsure if you can `exec` when piping... been meaning to look into that so LMK if you find out

**Comment by jaredallard** [comment]
Eh, just realized, this is calling `logfmt` which is in `eng` which is closed source. We might want to pull that out and OSS it otherwise `devspace` won't work for peepz

File: root/Makefile (hunk 2 above)
Comment (plisy) [reply root/Makefile:164]: They are different - in terms of running devconfig stuff.

File: shell/air-runner.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (plisy) [reply shell/air-runner.sh:35]: you can 👍 this seems to work just fine

[the remark this answers is not in the document]
Comment (plisy) [reply shell/air-runner.sh:35]: And yeah, the -a does the append part

File: root/Makefile [hunk 4]
```diff
@@ -34,6 +34,7 @@ FLY                 ?= $(shell ./scripts/shell-wrapper.sh gobin.sh -p github.com
 BENCH_FLAGS         := "-bench=Bench $(BENCH_FLAGS)"
 TEST_TAGS           ?= or_test,or_int
 SKIP_VALIDATE       ?=
+LOGFMT				?=
```
Comment (jaredallard) [inline root/Makefile:37]:
```suggestion
TEST_TAGS           ?= or_test,or_int
SKIP_VALIDATE       ?=
LOGFMT              ?=
```

**Review by jaredallard — APPROVED** [review]

**Pull request merged**